network/benches/socket_bench.rs

```diff
@@ -306,5 +307,80 @@ fn socket_bench(c: &mut Criterion) {
     c.bench("socket_send_throughput", bench);
 }
 
-criterion_group!(network_benches, socket_bench);
+/// Concurrently dial into server
+fn bench_client_connection<F, T, S>(
+    b: &mut Bencher,
+    concurrency: u64,
+    transport_func: F,
+    server_addr: NetworkAddress,
+) where
+    F: Fn() -> T,
+    T: Transport<Output = S> + 'static,
+    S: AsyncRead + AsyncWrite + Unpin + Send + 'static,
+{
+    let mut runtime = Builder::new()
+        .threaded_scheduler()
+        .core_threads(32)
```

Comment:
do we need `.core_threads(32)`? i thought tokio automatically spun up worker threads for each core?

Reply:
I think tokio is using # of physical cores